blood/src/main/java/cz/cuni/mff/d3s/blood/dependencyMatrix/DependencyMatrixCollector.java

```diff
@@ -1,36 +1,56 @@
 package cz.cuni.mff.d3s.blood.dependencyMatrix;
 
+import cz.cuni.mff.d3s.blood.utils.CheckedConsumer;
 import cz.cuni.mff.d3s.blood.utils.Result;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
 import org.graalvm.compiler.graph.Node;
 import org.graalvm.compiler.nodes.StructuredGraph;
 
 import java.io.PrintWriter;
 import java.io.StringWriter;
 import java.lang.reflect.Field;
 import java.util.Arrays;
+import java.util.Date;
+import java.util.HashSet;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.graalvm.compiler.phases.BasePhase;
 
 public final class DependencyMatrixCollector {
-    static ConcurrentHashMap<Class, ConcurrentHashMap<Class, DependencyValue>> dependencyTable = new ConcurrentHashMap<>();
+
+    private static final Predicate<Object> NON_NULL = Predicate.isEqual(null).negate();
+
+    // the default of 16 doesn't fit even the most trivial programs
+    private static final int HASHMAP_INIT_CAPACITY = 64;
+
+    static ConcurrentHashMap<Class, ConcurrentHashMap<Class, DependencyValue>> dependencyTable = new ConcurrentHashMap<>(HASHMAP_INIT_CAPACITY);
 
     public static void init() {
         // make sure, that Graal will track node creation position
         var props = System.getProperties();
         props.setProperty("debug.graal.TrackNodeCreationPosition", "true");
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            dump();
+        }, "dump at exit"));
     }
 
     private static Result<StackTraceElement[], String> getCreationStackTrace(Node node) {
-        Object a = node.getCreationPosition();
-        if (a == null) {
+        Object /*NodeCreationStackTrace*/ position = node.getCreationPosition();
```

Comment:
We should explain more, why we are assigning to Object instead of typed variable...